mutation/mutation_partition_serializer.cc

```diff
@@ -145,13 +147,16 @@ auto write_row_marker(Writer&& writer, const row_marker& marker)
 
 }
 
-template <typename RowTombstones>
+template <bool Preemptible, typename RowTombstones>
 static void write_tombstones(const schema& s, RowTombstones& row_tombstones, const range_tombstone_list& rt_list)
 {
     for (auto&& rte : rt_list) {
         auto& rt = rte.tombstone();
         row_tombstones.add().write_start(rt.start).write_tomb(rt.tomb).write_start_kind(rt.start_kind)
             .write_end(rt.end).write_end_kind(rt.end_kind).end_range_tombstone();
+        if (Preemptible && need_preempt()) {
+            yield().get();
```

Comment:
Could use maybe_yield() instead of need_preempt().

Reply:
And use `if constexpr` to highlight that it's compile-time.

Reply:
> Could use maybe_yield() instead of need_preempt().

yes, good idea

> And use if constexpr to highlight that it's compile-time.

Sure, no problem.